Terminal.Gui/Core/View.cs

```diff
@@ -1328,12 +1331,31 @@ public virtual void Redraw (Rect bounds)
 				textFormatter?.Draw (ViewToScreen (Bounds), HasFocus ? ColorScheme.Focus : ColorScheme.Normal, HasFocus ? ColorScheme.HotFocus : ColorScheme.HotNormal);
 			}
 
+			if (IsCurrentTop || this == Application.Top) {
+				if (!NeedDisplay.IsEmpty || LayoutNeeded) {
+					Driver.SetAttribute (Colors.TopLevel.Normal);
+
+					// This is the Application.Top. Clear just the region we're being asked to redraw 
+					// (the bounds passed to us).
+					Clear (bounds);
+					Driver.SetAttribute (Colors.Base.Normal);
+					((Toplevel)this).PositionToplevels ();
```

Comment:
I really don't like how this breaks the class hierarchy. A base class (`View`) should not, generally, have knowledge of a derived class (`TopLevel`). 

Do you think this is good practice?

Reply:
For a toplevel may be it is not so bad. If was the Window that is derived from Toplevel that is badly. But if you want I can revert that. So say something.

Reply:
Reverted. Thanks.